octobot_trading/exchanges/parser/funding_rate_parser.py

```diff
@@ -0,0 +1,151 @@
+                "never happen, check the parser code",
+                error=e,
+            )
+        self._create_debugging_report_for_record()
+        return self.formatted_record
+
+    def _parse_timestamp(self):
+        self._try_to_find_and_set(
+            FundingCols.TIMESTAMP.value,
+            [FundingCols.TIMESTAMP.value],
+            parse_method=parser_util.convert_any_time_to_seconds,
+            not_found_method=self.timestamp_not_found,
+        )
+
+    def timestamp_not_found(self, _):
+        try:
+            return int(self.exchange.connector.get_exchange_current_time())
+        except Exception as e:
+            self._log_missing(
+                FundingCols.TIMESTAMP.value,
+                f"{FundingCols.TIMESTAMP.value} not found and failed to get "
+                "time with get_exchange_current_time: ",
+                error=e,
+            )
+
+    def _parse_symbol(self):
+        self._try_to_find_and_set(
+            FundingCols.SYMBOL.value,
+            [FundingCols.SYMBOL.value],
+        )
+
+    def _parse_funding_rate(self):
+        self._try_to_find_and_set_decimal(
+            FundingCols.FUNDING_RATE.value,
+            [FundingCols.FUNDING_RATE.value] + FundingRateSynonyms.keys,
+            use_info_sub_dict=True,
+            allow_zero=True,
+        )
+
+    def _parse_predicted_funding_rate(self):
+        self._try_to_find_and_set_decimal(
+            FundingCols.PREDICTED_FUNDING_RATE.value,
+            [FundingCols.PREDICTED_FUNDING_RATE.value],
+            use_info_sub_dict=True,
+            not_found_val=trading_constants.NaN,
+            allow_zero=True,
+        )
+
+    def _parse_next_funding_time(self):
+        self._try_to_find_and_set(
+            FundingCols.NEXT_FUNDING_TIME.value,
+            [FundingCols.NEXT_FUNDING_TIME.value] + NextFundingTimeSynonyms.keys,
+            use_info_sub_dict=True,
+            parse_method=parser_util.convert_any_time_to_seconds,
+        )
+
+    def _parse_last_funding_time(self):
+        self._try_to_find_and_set(
+            FundingCols.LAST_FUNDING_TIME.value,
+            [FundingCols.LAST_FUNDING_TIME.value] + LastFundingTimeSynonyms.keys,
+            parse_method=parser_util.convert_any_time_to_seconds,
+            not_found_method=self.missing_last_funding_time,
+            use_info_sub_dict=True,
+        )
+
+    def missing_last_funding_time(self, _):
+        if next_funding := self.formatted_record.get(
+            FundingCols.NEXT_FUNDING_TIME.value
+        ):
+            return next_funding - self.FUNDING_TIME_UPDATE_PERIOD
+        self._log_missing(
+            FundingCols.LAST_FUNDING_TIME.value,
+            f"{FundingCols.LAST_FUNDING_TIME.value} not found, tried to "
+            f"calculate based on {FundingCols.NEXT_FUNDING_TIME.value} which is also missing",
+        )
+
+
+class FundingRateSynonyms:
+    keys = ["fundingRate"]
+
```

Comment:
I think we should only use tentacle to set those keys. There is no reason to have default value in generic code in my opinion

Reply:
I would highly suggest keeping the ones here that are safe.
as there is no downside and the upside is better support for potentially dozens of exchanges

Reply:
Here we are in generic code, in my opinion we cannot an **should not** assume the value of the key.

Reply:
Now its way better separated. Only the generic mode will try non default things.